MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,215 @@
+      properties.put("device", Build.DEVICE);
+      properties.put("tags", Build.TAGS);
+      properties.put("ISO03 language", Locale.getDefault().getISO3Language());
+      properties.put("language", Locale.getDefault().getLanguage());
+      properties.put("ISO03 country", Locale.getDefault().getISO3Country());
+      properties.put("country", Locale.getDefault().getCountry());
+      properties.put("display country", Locale.getDefault().getDisplayCountry());
+      properties.put("display name", Locale.getDefault().getDisplayName());
+      properties.put("display language", Locale.getDefault().getDisplayLanguage());
+      properties.put("size", isTablet ? IS_TABLET_MAP_VALUE : IS_PHONE_MAP_VALUE);
+
+      analytics.enqueue(TrackMessage.builder("New install")
+          .userId(loggedIn ? MAPBOX_USERNAME : "not logged in")
+          .properties(properties)
+      );
+    }
+  }
+
+  /**
+   * Makes an analytics call telling Segment which particular navigation drawer section has been selected.
+   *
+   * @param sectionName Name of the selected navigation drawer category
+   */
+  public void clickedOnNavDrawerSection(@NonNull String sectionName, boolean loggedIn) {
+    if (isAnalyticsEnabled()) {
+      trackEventWithProperties(CLICKED_ON_NAV_DRAWER_SECTION_EVENT_NAME, SECTION_NAME_MAP_KEY,
+          sectionName, loggedIn);
+    }
+  }
+
+  /**
+   * Makes an analytics call telling Segment which particular example has selected
+   *
+   * @param exampleName Name of the selected example
+   */
+  public void clickedOnIndividualExample(@NonNull String exampleName, boolean loggedIn) {
+    if (isAnalyticsEnabled()) {
+      trackEventWithProperties(CLICKED_ON_INDIVIDUAL_EXAMPLE_EVENT_NAME, EXAMPLE_NAME_MAP_KEY, exampleName, loggedIn);
+    }
+  }
+
+  /**
+   * Makes an analytics call telling Segment what custom-named event has happened. Because there are no
+   * custom parameters involved with the call, this method is the most "basic" analytics call that's
+   * available in this class.
+   *
+   * @param eventName Name of the event that's being recorded
+   */
+  public void trackEvent(@NonNull String eventName, boolean loggedIn) {
+    if (isAnalyticsEnabled()) {
+      trackEventWithProperties(eventName, null, null, loggedIn);
+    }
+  }
+
+  /**
+   * Makes an analytics call telling Segment what custom-named event has happened. Custom parameters provided
+   * are also included in the call.
+   *
+   * @param eventName             Name of the event that's being recorded
+   * @param keyForPropertiesMap   Key to the property being attached to the event that's being called
+   * @param valueForPropertiesMap Value of the property being attached to the event that's being called
+   */
+  public void trackEventWithProperties(@NonNull String eventName, String keyForPropertiesMap,
+                                       String valueForPropertiesMap, boolean loggedIn) {
+    if (isAnalyticsEnabled()) {
```

Comment:
You could do the opposite in these methods to reduce indentation: `if (!isAnalyticsEnabled()) return;`